agent/hcp/telemetry/gauge_store_test.go

```diff
@@ -0,0 +1,44 @@
+package telemetry
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func TestGaugeStore(t *testing.T) {
+	globalGauges := &gaugeStore{
```

Comment:
Maybe we could add a couple concurrency tests to validate any assumptions and to document your intent.

Reply:
Will resolve this and tag you in the new concurrent tests!